### DISCUSSION TITLE
Why do the key and value come from the prompt condition in MultiHeadCrossAttention?

### DISCUSSION THREAD

**Discussion by TtuHamg** [PixArt-alpha/PixArt-alpha#118] [Q&A]
It is an excellent work! But I notice that you hope query and value come from img tokens in [comment](https://github.com/PixArt-alpha/PixArt-alpha/blob/master/diffusion/model/nets/PixArt_blocks.py#L45). However, the code show that only query comes from img tokens, while key and value come from prompt conditions. Does this seem reasonable？
![image](https://github.com/PixArt-alpha/PixArt-alpha/assets/95615492/0fe4778b-625e-4c00-aa10-bd4342389b26)

**Comment by lawrence-cj** [comment]
The comment is wrong here. Will updated.

**Reply by TtuHamg** [reply]
Thank you for you reply. I have another question. I notice that the null condition sentence embedding (y_embedder.y_embedding) in your pretrained model (both 256 and 512 resolution) is same in each row except the first row. Is this specially designed? Hoping for your reply.
![image](https://github.com/PixArt-alpha/PixArt-alpha/assets/95615492/e462ac8a-452b-42db-ad03-58553813cbe9)

**Reply by lawrence-cj** [reply]
Maybe it's a little bug. I think it will not influence much. There is no design here.
